kernel/arm64/rot.c

```diff
@@ -0,0 +1,19 @@
+#include "common.h"
+
+#include "rot_kernel_sve.c"
```

Comment:
This should be guarded so as not to include SVE when it's not supported, see:
https://github.com/OpenMathLib/OpenBLAS/blob/develop/kernel/arm64/dot.c#L32-L40

Reply:
I tried adding the sve header guards, but after adding these header guards our sve kernels are not being called. 
Please help to rectify it.

Reply:
What did you add ? Did you copy the entire block that @Mousius marked, plus the actual `#ifdef HAVE_SVE...` include guard for your sve kernel ?
This should look something like
```
#include "common.h"
// Some compilers will report feature support for SVE without the appropriate
// header available
#ifdef HAVE_SVE
#if defined __has_include 
#if __has_include(<arm_sve.h>) && __ARM_FEATURE_SVE
#define USE_SVE
#endif 
#endif
#endif

#ifdef USE_SVE
#include "rot_kernel_sve.c"
#endif
```

Reply:
yes , I did the same.  I tried debugging the already sve implemented dot product also.   Even in dot product the "dot_kernel_sve.c" is not being called instead "dot_kernel_asimd.c" is called.

Reply:
Curious. Do you see HAVE_SVE in the compiler command lines, or in the generated config.h ?
Setting target ARMV8SVE should work, maybe autodetection did not pick the correct cpu (what kind of cpu are you building this on?)

Reply:
Nevermind, there is some broken logic in the autodetection code - right now HAVE_SVE will only be set when you specify a target. I'll produce a fix.

Reply:
It's in the generated config.h. I am using A64FX.

Reply:
does it work when you set TARGET=A64FX ?

Reply:
at least it works for me - you could try adding `#error using sve` after the `#ifdef USE_SVE` in either dot.c or your rot.c to see if compilation stops there. Obviously the other explanations would be that your toolchain lacks the arm_sve header or does not define the __ARM_FEATURE_SVE

Reply:
Yes. it works with TARGET=A64FX.